### PR TITLE
feat(login): allow logging in with presence data

### PR DESCRIPTION
**Pull request by Danktuary**
**Please describe the changes this PR makes and why it should be merged:**
There's currently no support for setting a presence upon login, which the Discord API does support (see: [this gateway identifying snippet](https://discordapp.com/developers/docs/topics/gateway#identifying)). This PR adds a 2nd parameter to `Client#login` and acts in the same manner as `ClientUser#setPresence`. This will keep users from making an extra API call inside their `ready` event just to set their initial status/activity, as so many of us do.

Example snippet:
```js
client.login(token, {
    presence: {
        status: 'dnd',
        activity: {
            name: 'YouTube',
            type: 'WATCHING',
        },
    },
});
```

There were some things I wasn't very sure of (e.g. `Client#login`'s return value and the code inside `identifyNew()`), so if anyone has any suggestions on betters way to go about those, I'll gladly comply.

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by KingDGrizzle** [comment]
This looks great, but I'd feel like it would look better in the Client constructor (so `new Client({ presence: {} })`)

That way login only handles the actual login, and we can still do this. But that's just my opinion. :smile:

File: src/client/Client.js [hunk 1]
```diff
@@ -238,22 +238,46 @@ class Client extends BaseClient {
...
+   * @example
+   * client.login('my token', {
+   *  presence: {
+   *    status: 'dnd',
+   *    activity: {
+   *      name: 'YouTube',
+   *      type: 'WATCHING',
+   *    },
+   *  },
+   * });
```
Comment (Lewdcario) [inline src/client/Client.js]: Add promise handling to be more consistent with other examples

File: src/client/ClientManager.js [hunk 2]
```diff
@@ -32,14 +32,18 @@ class ClientManager {
   /**
    * Connects the client to the WebSocket.
    * @param {string} token The authorization token
+   * @param {LoginOptions} options Options to log in with
    * @param {Function} resolve Function to run when connection is successful
    * @param {Function} reject Function to run when connection fails
    */
-  connectToWebSocket(token, resolve, reject) {
+  connectToWebSocket(token, options, resolve, reject) {
```
Comment (Lewdcario) [inline src/client/ClientManager.js]: `connectToWebSocket` should be async to avoid the later callback.

**Comment by bdistin** [comment]
+1 on KingDGrizzle's comment... Should be a client option...

**Comment by Danktuary** [comment]
@KingDGrizzle @bdistin Turns out it already a client option. Something like this will work (thanks to @Hackzzila for pointing it out 🎉):
```js
const client = new Client({
	ws: {
		presence: {
			status: 'idle',
			game: {
				name: 'testst',
				type: 0,
			},
		},
	},
});
```

So I'm just gonna close this PR since it wasn't an optimal implementation anyway, and make a different once that makes `WebsocketOptions#presence` accepts the same data as `ClientUser#setPresence` instead (as well as making it documented, since it's currently not).

**Pull request closed without merging**